lib/GenomeFileUtil/core/FastaGFFUploaderScript.py

```diff
@@ -0,0 +1,678 @@
+        #Upload FASTA to shock
+        #Need to gunzip file first
+        gunzipped_fasta_file=input_fasta_file
+#        gunzipped_fasta_file=input_fasta_file[0:-3]
+#        with gzip.open(input_fasta_file, 'rb') as f_in:
+#            with open(gunzipped_fasta_file, 'wb') as f_out:
+#                shutil.copyfileobj(f_in, f_out)
+
+        token = os.environ.get('KB_AUTH_TOKEN') 
+        fasta_handle_ref=None
+        shock_info = script_utils.upload_file_to_shock(logger, shock_service_url, gunzipped_fasta_file, token=token)
+        shock_id = shock_info["id"]
+        handles = script_utils.getHandles(logger, shock_service_url, handle_service_url, [shock_id], [fasta_handle_ref], token)
+        fasta_handle_ref=handles[0]
+        assembly['fasta_handle_ref'] = fasta_handle_ref
+        
+        #Remove gunzipped file
+        #os.remove(input_fasta_file[0:-3])
+
+        #Provenance has a 1 MB limit.  We may want to add more like the accessions, but to be safe for now not doing that.
+        assembly_provenance = [{"script": __file__, "script_ver": "0.1", "description": "Assembly from FASTA from %s" % (source)}]
+
+        #Report large genomes?
+        hidden=0
+
+        logger.info("Attempting Assembly save for %s" % (assembly["assembly_id"]))
+        assembly_not_saved = True
+        while assembly_not_saved:
+            try:
+                assembly_info =  ws_client.save_objects({"workspace":workspace_name,
+                                                           "objects":[ { "type": "KBaseGenomeAnnotations.Assembly",
+                                                                         "data": assembly,
+                                                                         "name": assembly["assembly_id"],
+                                                                         "hidden" : hidden,
+                                                                         "provenance": assembly_provenance}]}) 
+                assembly_not_saved = False 
+                logger.info("Assembly saved for %s" % (assembly["name"]))
+                assembly_ref = workspace_name+"/"+assembly["assembly_id"]
+            except WorkspaceError as e:
+                logger.info('Logging workspace error on save_objects: {}\n{}'.format(e.message, e.data))
+                raise
+
+    genome = dict()
+    genome["id"]=core_genome_name
+    genome["scientific_name"]=scientific_name
+    genome["assembly_ref"]=assembly_ref
+    genome["features"]=genome_features_list
+    genome["cdss"]=genome_cdss_list
+    genome["mrnas"]=genome_mrnas_list
+    genome["source"]=source
+    genome["domain"]="Eukaryota"
+    genome["genetic_code"]=1
+    genome["gc_content"]=assembly["gc_content"]
+    genome["dna_size"]=assembly["dna_size"]
+
+    if taxon_reference is not None:
+        genome["taxon_ref"] = taxon_reference
+        genome["taxonomy"]=taxonomy
+
+    UserMeta = dict()
+    UserMeta['Taxonomy']=taxonomy
```

Comment:
You shouldn't set any metadata fields that are automatically extracted by the WS.  It looks like most of these are extracted for you.  I forget whether this will overwrite what is extracted or do nothing, but still we should remove any code that attempts to set them.